airflow/providers/cncf/kubernetes/executors/kubernetes_executor.py

```diff
@@ -199,13 +200,32 @@ def clear_not_launched_queued_tasks(self, session: Session = NEW_SESSION) -> Non
             assert self.kube_client
         from airflow.models.taskinstance import TaskInstance
 
+        has_executor = hasattr(TaskInstance, "executor")
+        default_executor = None
+        if has_executor:
+            from airflow.executors.executor_loader import ExecutorLoader
+
+            default_executor = str(ExecutorLoader.get_default_executor_name())
+
         with Stats.timer("kubernetes_executor.clear_not_launched_queued_tasks.duration"):
             self.log.debug("Clearing tasks that have not been launched")
+
             query = select(TaskInstance).where(
-                TaskInstance.state == TaskInstanceState.QUEUED, TaskInstance.queued_by_job_id == self.job_id
+                TaskInstance.state == TaskInstanceState.QUEUED,
+                TaskInstance.queued_by_job_id == self.job_id,
             )
             if self.kubernetes_queue:
                 query = query.where(TaskInstance.queue == self.kubernetes_queue)
+            elif has_executor and KUBERNETES_EXECUTOR == default_executor:
```

Comment:
Shouldn't this just be an if? It's not going to get evaluated if `self.kubernetes_queue` exists, and those don't feel mutually exclusive to me?

Reply:
This exclusive for celery kubernetes executor (deprecated) so if its that case no need to check for hybrid executor